src/parsers/manifest/dash/parsePeriods.ts

```diff
@@ -405,18 +405,44 @@ export default function parsePeriods(
           if (audioDescription != null) {
             parsedAdaptationSet.audioDescription = audioDescription;
           }
+
           const parsedAdaptation = parsedAdaptations[type];
+
           if (!parsedAdaptation) {
             parsedAdaptations[type] = [parsedAdaptationSet];
             if (isMainAdaptation && type === "video") {
               acc.videoMainAdaptation = parsedAdaptationSet;
             }
-          } else if (isMainAdaptation && type === "video") {
-            // put "main" adaptation as the first
-            parsedAdaptation.unshift(parsedAdaptationSet);
-            acc.videoMainAdaptation = parsedAdaptationSet;
           } else {
-            parsedAdaptation.push(parsedAdaptationSet);
+            if (adaptation.children.supplementalProperties != null) {
+              const adaptationSetSwitchingIDs : string[] = [];
+              const { supplementalProperties } = adaptation.children;
+              for (let j = 0; j < supplementalProperties.length; j++) {
+                const supplementalProperty = supplementalProperties[j];
+                if (
+                  supplementalProperty.schemeIdUri ===
+                  "urn:mpeg:dash:adaptation-set-switching:2016" &&
+                  supplementalProperty.value != null
+                ) {
+                  const ids = supplementalProperty.value.split(",")
+                    .map(id => id.trim())
+                    .filter(id => id);
+                  if (ids.length) {
+                    adaptationSetSwitchingIDs.push(...ids);
+                  }
+                }
+              }
+              if (adaptationSetSwitchingIDs.length) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Either remove or fill this block of code. ([typescript:S108](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS108))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=410&issues=AWeEvHho8YaV-M4wKbC8&open=AWeEvHho8YaV-M4wKbC8)

[ISSUE_KEY]: AWeEvHho8YaV-M4wKbC8